File: backend/events.ts
export interface MeshEvent {
  timestamp: string;
  event_type: string;
  message: string;
  affected_node: string | null;
  severity: string;
}

export const EVENT_LOG: MeshEvent[] = [];

function timestamp(): string {
  const d = new Date();
  const pad = (n: number) => String(n).padStart(2, '0');
  return `${pad(d.getUTCHours())}:${pad(d.getUTCMinutes())}:${pad(d.getUTCSeconds())}`;
}

export function addEvent(
  eventType: string,
  message: string,
  affectedNode: string | null = null,
  severity = 'info'
): MeshEvent {
  const event: MeshEvent = {
    timestamp: timestamp(),
    event_type: eventType,
    message,
    affected_node: affectedNode,
    severity,
  };
  EVENT_LOG.unshift(event);
  return event;
}

export function getEvents(limit = 20): MeshEvent[] {
  return EVENT_LOG.slice(0, limit);
}

File: backend/events.py
"""Simple in-memory event log for network activity."""

from __future__ import annotations

from datetime import datetime, timezone

EVENT_LOG = []


def _timestamp():
    return datetime.now(timezone.utc).strftime("%H:%M:%S")


def add_event(event_type, message, affected_node=None, severity="info"):
    event = {
        "timestamp": _timestamp(),
        "event_type": event_type,
        "message": message,
        "affected_node": affected_node,
        "severity": severity,
    }
    EVENT_LOG.insert(0, event)
    return event


def get_events(limit=20):
    return list(EVENT_LOG[:limit])
